pandas/tests/indexes/timedeltas/test_arithmetic.py

```diff
@@ -453,10 +453,16 @@ def test_timedelta_ops_with_missing_values(self):
         # setup
         s1 = pd.to_timedelta(Series(['00:00:01']))
         s2 = pd.to_timedelta(Series(['00:00:02']))
-        sn = pd.to_timedelta(Series([pd.NaT]))
+        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
+            # Passing datetime64-dtype data to TimedeltaIndex is deprecated
+            sn = pd.to_timedelta(Series([pd.NaT]))
```

Comment:
this is not correct 
it is all nan which is not definitive

Reply:
`Series([pd.NaT])` has `datetime64[ns]` dtype.  After we've done inference, we need to stick with it.  I can make this test clearer by putting the dtype in the Series constructor explicitly

Reply:
what i mean is that this is all nan and inference is ambiguous at this point. so I wouldn't expect the warning (unless the dtype IS explicity passed). Can address in a followup, but pls open an issue, this is IMHO a bug.